src/main/cheffy/recipes.clj

```diff
@@ -58,12 +59,37 @@
         ;; - see response-for call in dev.clj
         account-id (get-in request [:headers "authorization"])
         recipes (query-recipes db account-id)]
-    (-> recipes json/generate-string response)))
+    (-> recipes json/generate-string response/response)))
 
+(defn create-recipe! [conn account-id
+                      {:keys [name public prep-time img] :as _params}]
+  ;; TODO: change UUID later to something better indexed by Datomic
+  (let [recipe-id (random-uuid)]
+    (d/transact conn {:tx-data [{:recipe/recipe-id recipe-id
+                                 :recipe/display-name name ; shadowing core function
+                                 :recipe/public? public
+                                 :recipe/prep-time prep-time
+                                 :recipe/image-url img
+                                 ;; owner is actually a ref
+                                 :recipe/owner [:account/account-id account-id]}]})
+    recipe-id))
 
-;; TODO juraj: is this really needed?
+
+(defn create-recipe-response
+  ;; TODO: why do we use `transit-params` and not just `params`
+  ;; - this makes the code transport specific
```

Comment:
`:transit-params` is specific key added by one of the core's interceptors - we cannot change that.